### PR TITLE
update DOT sdk

### PR DESCRIPTION
**Pull request by noel-bitgo**
update DOT library to remove controller optional field -> https://github.com/paritytech/txwrapper-core/pull/309 and https://github.com/paritytech/substrate/pull/14039

DOT is defaulting the controller address as the stash address

### CODE REVIEW

File: modules/sdk-coin-dot/src/lib/batchTransactionBuilder.ts [hunk 1]
```diff
@@ -207,12 +207,10 @@ export class BatchTransactionBuilder extends TransactionBuilder {
     return unsigned.method;
   }
 
-  private getBondCall(args: StakeBatchCallArgs): string {
+  public getBondCall(args: StakeBatchCallArgs): string {
```
Comment (noel-bitgo) [inline modules/sdk-coin-dot/src/lib/batchTransactionBuilder.ts:210]: just to test

**Review by zahin-mohammad — COMMENTED** [review]
PR is not ready for review

File: modules/sdk-coin-dot/src/lib/iface.ts [hunk 2]
```diff
@@ -179,7 +179,7 @@ export type StakeArgsPayeeRaw = { controller?: null; stash?: null; staked?: null
  */
 export interface StakeArgs {
   value: string;
-  controller?: { id: string };
+  controller: { id: string };
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-dot/src/lib/iface.ts:182]: Undocumented breaking change. Need a BREAKING CHANGE: <comment> in the commit footer.

**Review by mmcshinsky-bitgo — CHANGES_REQUESTED** [review]

File: modules/sdk-coin-dot/src/lib/iface.ts (hunk 2 above)
Comment (noel-bitgo) [reply modules/sdk-coin-dot/src/lib/iface.ts:182]: this is actually reverting the previous change https://github.com/BitGo/BitGoJS/commit/a6f6a0d96afdb663a7a096c4eb2caf9ca1649d63

Comment (noel-bitgo) [reply modules/sdk-coin-dot/src/lib/iface.ts:182]: i'll keep that in mind for future breaking changes

**Review by OttoAllmendinger — COMMENTED** [review]
I have insufficient context to review this PR.

To flush it from my review queue, I am leaving this comment :ocean: 

This PR will still appear in the review queue of the other team members.

**Review by zahin-mohammad — COMMENTED** [review]
Im lacking some context on the dot implementation, but from skimming the PR it lgtm. If you can get an approval from eth-alt or staking I'll approve this PR.

**Review by sachushaji — APPROVED** [review]

**Review by aarvay — APPROVED** [review]

**Review by alebusse — APPROVED** [review]

**Review by mmcshinsky-bitgo — APPROVED** [review]

**Pull request merged**